src/GovUk.Education.ExploreEducationStatistics.Publisher/Functions/PublishReleaseContentFunction.cs

```diff
@@ -83,17 +81,24 @@ public class PublishReleaseContentFunction
 
                 // Invalidate the cached trees in case any methodologies/publications
                 // are now accessible for the first time after publishing these releases
-                await _blobCacheService.DeleteItem(new AllMethodologiesCacheKey());
-                await _blobCacheService.DeleteItem(new PublicationTreeCacheKey());
-                await _blobCacheService.DeleteItem(new PublicationTreeCacheKey(PublicationTreeFilter.AnyData));
-                await _blobCacheService.DeleteItem(new PublicationTreeCacheKey(PublicationTreeFilter.LatestData));
+                await _contentService.DeleteCachedTaxonomyBlobs();
 
+                // Invalidate publication cache for release
                 var release = await _contentDbContext.Releases
                     .Include(r => r.Publication)
                     .Where(r => r.Id == message.ReleaseId)
                     .SingleAsync();
                 await _blobCacheService.DeleteItem(new PublicationCacheKey(release.Publication.Slug));
-                // TODO: @MarkFix EES-3149 Delete superseded publication's cache here too?
+
+                // Invalidate publication cache for superseded publications, as potentially affected. If newly
+                // published release is first Live release for the publication, the superseding is now enforced
+                var supersededPublications = await _contentDbContext.Publications
+                    .Where(p => p.SupersededById == release.Publication.Id)
+                    .ToListAsync();
+                foreach (var p in supersededPublications)
+                {
+                    await _blobCacheService.DeleteItem(new PublicationCacheKey(p.Slug));
+                }
```

Comment:
You could use `ForEachAwaitAsync` like

```
                await _contentDbContext.Publications
                    .Where(p => p.SupersededById == release.Publication.Id)
                    .ToAsyncEnumerable()
                    .ForEachAwaitAsync(publication =>
                        _blobCacheService.DeleteItem(new PublicationCacheKey(publication.Slug)));
```

You could also refactor the blob service to be able to delete multiple cache keys at once.

Reply:
I've just used `ForEachAwaitAsync` rather than refactored the blob service. I reached for `foreach` here because of that previous issue with the `ForEachAwaitAsync` stuff and now I don't trust it. But if you're happy that's good here I'll go with that.